### PR TITLE
Add support for the PLACEMENT KEY column constraint in Spanner import/export templates

### PR DESCRIPTION
**Pull request by sgorse123**
As part of geo-partitioning, users can define Placement Tables by creating a table with a specific column that uses the PLACEMENT KEY constraint ([Example](https://cloud.google.com/spanner/docs/create-manage-data-placements#create-table))

The following change exports and imports the PLACEMENT KEY constraint similar to how other column constraints are imported/exported (NOT NULL, STORED, etc.).

Also manually validated by creating a database with placement tables, and running the updated import and export templates.

### CODE REVIEW

**Review by Deep1998 — DISMISSED** [review]
LGTM, ITs failing seems unrelated.

**Review by Deep1998 — APPROVED** [review]

**Comment by Deep1998** [comment]
@sgorse123 the test `ExportPipelineIT.testSpannerToGCSAvroStaging` seems to be repeatedly failing. Could you take a look?

**Comment by sgorse123** [comment]
TLDR: My current thought is this failure is unrelated to my change.

Relevant error log:
```
INVALID_ARGUMENT: Operation with name "projects/cloud-teleport-testing/instances/teleport/databases/testspa_20240821_161855_xana0j/operations/r915617f6_5865_41de_ad92_e893e32531f2" failed with status = GrpcStatusCode{transportCode=INVALID_ARGUMENT} and message = Token key MyTokens in search index testSpannerToGCSAvroStaging_SearchIndex does not support order
```

Looks like this is due to the following DDL statement failing: https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/b3c32a7b2c2572ea3e56909c742008e9ee0a06c5/v1/src/test/java/com/google/cloud/teleport/spanner/ExportPipelineIT.java#L200-L205

I tried to repro this locally on my branch, and the test passed:
```
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,956.715 s - in com.google.cloud.teleport.spanner.ExportPipelineIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  33:45 min
[INFO] Finished at: 2024-08-22T05:37:34Z
[INFO] ------------------------------------------------------------------------
```

**Comment by Deep1998** [comment]
Agreed, looks like it failing for other PRs as well.

**Comment by Polber** [comment]
@Deep1998 These changes seem to be breaking release: https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/10509501447

You can download test report at bottom to see errors. Latest error seems to be:
```
Caused by: com.google.cloud.spanner.SpannerException: INVALID_ARGUMENT: Operation with name "projects/cloud-teleport-testing/instances/teleport/databases/testspa_20240821_152954_tvex6w/operations/r559fb3f2_87d5_4507_a74f_7042717a1221" failed with status = GrpcStatusCode{transportCode=INVALID_ARGUMENT} and message = Token key MyTokens in search index testSpannerToGCSAvroStaging_SearchIndex does not support order.
```

`testSpannerToGCSAvroStaging` is still failing (and other Import/Export pipeline IT's seem to be flaky or failing when run against Staging Spanner)

**Comment by Deep1998** [comment]
@Polber this test has been failing before this PR was merged ([Example](https://github.com/GoogleCloudPlatform/DataflowTemplates/runs/28973228732)). The failure seems to be unrelated to this change.

**Comment by Deep1998** [comment]
I can take a look later if a PR caused this breaking change earlier or whether its some spanner staging rollout for which this fails.

**Comment by Polber** [comment]
It seems like all the flakes and failures have been strictly on Spanner staging tests specifically
